sentry/src/main/java/io/sentry/SentryTracer.java

```diff
@@ -0,0 +1,245 @@
+  }
+
+  @Override
+  public boolean finish() {
+    return this.finish(this.getStatus());
+  }
+
+  @Override
+  public boolean finish(@Nullable SpanStatus status) {
+    if (root.finish(status)) {
+      hub.configureScope(
+          scope -> {
+            scope.withTransaction(
+                transaction -> {
+                  if (transaction == this) {
+                    scope.clearTransaction();
+                  }
+                });
+          });
+      SentryTransaction transaction = new SentryTransaction(this);
+      hub.captureTransaction(transaction);
+      return true;
+    } else {
+      return false;
+    }
+  }
+
+  @Override
+  public void setOperation(final @NotNull String operation) {
+    this.root.setOperation(operation);
+  }
+
+  @Override
+  public @NotNull String getOperation() {
+    return this.root.getOperation();
+  }
+
+  @Override
+  public void setDescription(final @Nullable String description) {
+    this.root.setDescription(description);
+  }
+
+  @Override
+  public @Nullable String getDescription() {
+    return this.root.getDescription();
+  }
+
+  @Override
+  public void setStatus(final @Nullable SpanStatus status) {
+    this.root.setStatus(status);
+  }
+
+  @Override
+  public @Nullable SpanStatus getStatus() {
+    return this.root.getStatus();
+  }
+
+  @Override
+  public void setThrowable(final @Nullable Throwable throwable) {
+    this.root.setThrowable(throwable);
+  }
+
+  @Override
+  public @Nullable Throwable getThrowable() {
+    return this.root.getThrowable();
+  }
+
+  @Override
+  public @NotNull SpanContext getSpanContext() {
+    return this.root.getSpanContext();
+  }
+
+  @Override
+  public void setTag(final @NotNull String key, final @NotNull String value) {
+    this.root.setTag(key, value);
+  }
+
+  @Override
+  public @Nullable String getTag(final @NotNull String key) {
+    return this.root.getTag(key);
+  }
+
+  @Override
+  public boolean isFinished() {
+    return this.root.isFinished();
+  }
+
+  @Override
+  public @Nullable Boolean isSampled() {
+    return this.root.isSampled();
+  }
+
+  @Override
+  public void setName(@NotNull String name) {
+    this.root.setTag(ISpan.NAME_TAG, name);
+  }
+
+  @Override
+  public @NotNull String getName() {
+    return this.root.getTag(ISpan.NAME_TAG);
+  }
+
+  @Override
+  @Deprecated
+  @ApiStatus.ScheduledForRemoval
+  public void setRequest(@Nullable Request request) {
+    hub.configureScope(scope -> scope.setRequest(request));
+  }
+
+  @Override
+  @Deprecated
+  @ApiStatus.ScheduledForRemoval
+  public @Nullable Request getRequest() {
+    final AtomicReference<Request> contexts = new AtomicReference<>();
+    hub.configureScope(scope -> contexts.set(scope.getRequest()));
+    return contexts.get();
+  }
+
+  @Override
+  @Deprecated
+  @ApiStatus.ScheduledForRemoval
+  public @NotNull Contexts getContexts() {
+    final AtomicReference<Contexts> contexts = new AtomicReference<>();
```

Comment:
This can have surprising side effects. `Contexts` is thread-safe so we could have a instance here and set it to the SentryTransaction when we capture it.

IMHO changing this instance having side effects on the scope isn't the way to go.